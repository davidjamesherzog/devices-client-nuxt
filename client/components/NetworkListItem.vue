<template>
  <div class="box" :data-list-item-id="network.id">
    <article class="media">
      <div class="media-left">
        <div class="content has-text-centered">
          <b-icon
            :icon="type"
            size="is-large"
            type="is-primary"
            data-card-icon
          />
        </div>
      </div>
      <div class="media-content">
        <div class="content">
          <p>
            <strong :data-list-item-title="network.id">{{
              network.name
            }}</strong>
            <small :data-list-item-type="network.id">{{ network.type }}</small>
            <br />
            <span :data-list-item-description="network.id">
              {{ network.description }}
            </span>
          </p>
        </div>
      </div>
    </article>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator';
import Network from '~/client/models/Network';

@Component({
  name: 'NetworkList'
})
export default class NetworkList extends Vue {
  @Prop({ type: Object })
  private network!: Network;

  get type(): String {
    return this.network.type === 'WIFI' ? 'wifi' : 'ethernet-cable';
  }
}
</script>

<style lang="scss" scoped>
.box {
  margin-bottom: 0.5rem;
  cursor: pointer;
}
</style>
