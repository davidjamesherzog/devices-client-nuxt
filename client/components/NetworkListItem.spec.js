import { createLocalVue, shallowMount } from '@vue/test-utils';
import Buefy from 'buefy';
import NetworkListItem from '~/components/NetworkListItem.vue';

describe('NetworkListItem', () => {
  describe('type', () => {
    test('should return wifi', () => {
      const networkListItem = new NetworkListItem();
      networkListItem.network = {
        type: 'WIFI'
      };
      expect(networkListItem.type).toEqual('wifi');
    });

    test('should return ethernet-cable', () => {
      const networkListItem = new NetworkListItem();
      networkListItem.network = {
        type: 'ETHERNET'
      };
      expect(networkListItem.type).toEqual('ethernet-cable');
    });
  });

  describe('template', () => {
    let localVue;

    beforeEach(() => {
      localVue = createLocalVue();
      localVue.use(Buefy);
    });

    test('should render network list item', () => {
      const wrapper = shallowMount(NetworkListItem, {
        localVue,
        propsData: {
          network: {
            id: 1,
            name: 'TestWifi',
            description: 'Test description',
            type: 'WIFI'
          }
        }
      });
      const id = wrapper.find('[data-list-item-id="1"]');
      expect(id).toBeDefined();
      const title = wrapper.find('[data-list-item-title="1"]');
      expect(title.text()).toEqual('TestWifi');
      const description = wrapper.find('[data-list-item-description="1"]');
      expect(description.text()).toEqual('Test description');
      const type = wrapper.find('[data-list-item-type="1"]');
      expect(type.text()).toEqual('WIFI');
      expect(wrapper).toMatchSnapshot();
    });
  });
});
