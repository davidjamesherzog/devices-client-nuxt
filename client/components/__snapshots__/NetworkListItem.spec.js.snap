// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkListItem template should render network list item 1`] = `
<div data-list-item-id="1" class="box">
  <article class="media">
    <div class="media-left">
      <div class="content has-text-centered">
        <b-icon-stub type="is-primary" icon="wifi" size="is-large" data-card-icon=""></b-icon-stub>
      </div>
    </div>
    <div class="media-content">
      <div class="content">
        <p><strong data-list-item-title="1">TestWifi</strong> <small data-list-item-type="1">WIFI</small> <br> <span data-list-item-description="1">
            Test description
          </span></p>
      </div>
    </div>
  </article>
</div>
`;
